app/lib/redcarpet/render/html_rouge.rb

```diff
@@ -14,7 +14,11 @@ def link(link, _title, content)
         @options[:link_attributes]&.each do |attribute, value|
           link_attributes += %( #{attribute}="#{value}")
         end
-        %(<a href="#{link}"#{link_attributes}>#{content}</a>)
+        if (/A(https?:\/\/)/.match? link) || link.nil?
```

Comment:
I think it should be `\A`

Reply:
Another thing that came to my mind while passively thinking about this was, how should we handle in case only `https://` or `http://` is passed? Since the regex would get a positive for the same. Should we append a `.` or `.+\Z` at the end after `\/` and add another `elsif` which would be checking setting `<a href="">` in case it is only `https://` or `http://`?

Reply:
Oops, thanks for catching that @rhymes !

@Amorpheuz nice catch but I don't think we need to account for all possible invalid link cases. it not a major issue If given link is invalid. The bigger issue, which this PR address, is accidentally assuming a given link is a relative link.

Side note: GitHub has the same issue 😄